app/controllers/support_interface/application_forms/courses_controller.rb

```diff
@@ -45,6 +45,32 @@ def create
         end
       end
 
+      def confirm_change
+        @change_course_choice = ChangeCourseChoiceForm.new
+      end
+
+      def change
+        @change_course_choice = ChangeCourseChoiceForm.new(change_course_choice_params)
+
+        begin
+          if @change_course_choice.save(params[:application_choice_id])
+            flash[:success] = 'Course successfully changed'
+            redirect_to support_interface_application_form_path(application_form_id)
+          else
+            render :confirm_change
+          end
+        rescue ActiveRecord::RecordNotFound
```

Comment:
These (quite generic) errors could be rescued in `save` and re-raised as a specific error type like `CourseChangeError` with the appropriate error message. It would keep the error message logic closer to the scene of the crime. 
It's also easier to test for a specific error type.

Reply:
🕵️